CorsixTH/Src/th_gfx_sdl.cpp

```diff
@@ -722,7 +724,69 @@ void render_target::set_cursor_position(int iX, int iY) {
   cursor_y = iY;
 }
 
-bool render_target::take_screenshot(const char* sFile) {
+namespace {
+
+bool write_rgb_png(int width, int height, png_bytep pixels, int pitch,
+                   const char* file_path) {
+  // Open the file for writing
+  FILE* fp = std::fopen(file_path, "wb");
```

Comment:
You don't use this file until much later in png_init_io but need to handle it in the error case many times before that, it likely makes sense to delay opening the file until later.

Reply:
The libpng document suggested to open the file first, in order to avoid committing "considerable resources" to PNG writing if the file cannot be opened, but may be old advice.

Reply:
Moved.